File: 0273_Modest - 2013 - Radiative heat transfer/代码/第三版/chapter_02_03/emdiel.cpp
// Function Emdiel

# include <iostream.h>
# include <math.h>
	
double emdiel(double n)
	{
	// Calculates spectral, hemispherical emittance of dielectric, eq(3.82)
	double emdiel;
	emdiel=(4.E0*n+2.E0)/3.E0/pow((n+1.E0),2)+2.E0*pow(n,3)*(n*n+2.E0*n-1.E0) 
		/(n*n+1.E0)/(pow(n,4)-1.E0)-pow((n*(n*n-1.E0)),2)/pow((n*n+1.E0),3)
		*log((n-1.E0)/(n+1.E0)) -8.E0*pow(n,4)*(pow(n,4)+1.E0)*log(n)/
		(n*n+1.E0)/pow((pow(n,4)-1.E0),2);
	return emdiel;
	}

File: 0273_Modest - 2013 - Radiative heat transfer/代码/第四版/Chapters本书/Chapter02-03/emdiel.f90
    function emdiel(n)
! Calculates spectral, hemispherical emittance of dielectric, eq(3.82)
    real*8 emdiel,n
    emdiel=(4.d0*n+2.d0)/3.d0/(n+1.d0)**2+2.d0*n**3*(n*n+2.d0*n-1.d0) &
                    /(n*n+1.d0)/(n**4-1.d0) &
                    -(n*(n*n-1.d0))**2/(n*n+1.d0)**3*dlog((n-1.d0)/(n+1.d0)) &
                    -8.d0*n**4*(n**4+1.d0)*dlog(n)/(n*n+1.d0)/(n**4-1.d0)**2
    return
    end
